src/impl/pooling_direct.jl

```diff
@@ -78,11 +83,17 @@ for name in (:max, :mean)
                     end
                 elseif $(name == :mean)
                     m += x[input_kw, input_kh, input_kd, c, batch_idx]
+                elseif $(name == :lp)
+                    # y = (∑ x^p)^(1/p), here to calculate (∑ x^p)
+                    m += x[input_kw, input_kh, input_kd, c, batch_idx]^p
                 else
                     error("Unimplemented codegen path")
                 end
             end
 
+            # for lppool, y = (∑ x^p)^(1/p)
+            m = $(name == :lp) ? m^(1 / p) : m
```

Comment:
```suggestion
            m = $(name == :lp) ? m^(T(1) / p) : m
```
We don't want to cast Float64 here. I think we should also put some type check or conversion on `p` somewhere above.